src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java

```diff
@@ -0,0 +1,93 @@
+package com.github.princesslana.eriscasper.repository.event;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.event.*;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import com.github.princesslana.eriscasper.data.resource.ImmutableChannel;
+import com.github.princesslana.eriscasper.repository.GuildRepository;
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+
+public class GuildsFromEvents implements GuildRepository {
+
+  private final Map<Snowflake, Guild> guildCache;
+  private final Map<Snowflake, Channel> channelCache;
+
+  public GuildsFromEvents(Observable<Event> eventObservable) {
+    guildCache = new HashMap<>();
+    channelCache = new HashMap<>();
+    eventObservable
+        .ofType(GuildCreateEvent.class)
+        .map(GuildCreateEvent::unwrap)
+        .subscribe(
+            guild -> {
+              guildCache.putIfAbsent(guild.getId(), guild);
```

Comment:
Why are we using `putIfAbsent`? If we've received another `GuildCreateEvent` with the same id, wouldn't the data from the new event be more up to date?